wasi/2022/WASI-04-21.md

```diff
@@ -31,3 +31,64 @@ The meeting will be on a zoom.us video conference.
         1. More design overview
         2. Poll to advance to next phase
     2. _Sumbit a PR to add your agenda item here_
+
+## Notes
+### Attendees
+- Lin Clark
+- Till Schneidereit
+- George Kulakowski
+- Dave Bakker
+- Nathaniel McCallum
+- Andrew Brown
+- Bailey Hayes
+- Brian Hardock
+- Dan Gohman
+- David Yamnitsky
+- Granville Schmidt
+- Isabella Tromba
+- Johnnie Birch
+- Luke Wagner
+- Mohamad Jaafar Nehme
+- Piotr Sikora
+- Radu Matei
+- Sebastien Deleuze
+- Syrus Akbary
+- Yong He
+- Sam Clegg
+
+### wasi-url poll
+**George Kulakowski:** proposal to implement wasi-url set of interfaces. Parsing given a base url, make a new url, accessors into those. Will follow WHATWG specification for URLs. Motivations are that URLs are a common and ubiq interface. Many components will want to use or provide URLs
+
+Does anyone want to talk about anything before poll? 
+
+**Sam Clegg:** Not really that involved with webby stuff. URL vs URI?
+
+**George Kulakowski:** WHATWG takes position that URI vs URL has caused more confusion so they go with URL. URN is still a part of this spec. 
+
+**Sam Clegg:** Sounds like it’s been decided.
+
+**George Kulakowski:** WHATWG is starting point for this discussion. Would be interested in potential interop issues if they come up in the future, or crossing boundaries… if there’s ever been compat issues with that.
+
+**Sam Clegg:** Prob too early to talk about this kind of stuff, but will you attempt to mirror the JS API?
+
+**George Kulakowski:** Another good question. Right now, this is a starting point for other work. Having an interface at the boundary in WASI makes sense, but then you’d have implementations that go from interface you have to the intermediate representation. This is something we should address, but not in first version of this api.
+
+**Till Schneidereit:** I implemented JS API using Rust’s implementation and there are clearly API differences between these two languages. That holds across different ecosystems. We’ll not be able to generate the expected API surface for all of these ecosystems. Given how URL is constrained, you don’t have much design space. A lot of the differences are cosmetic in nature. The adapter you have to do to go from abstract to lang idiomatic should be small.
+
+One example where JS URL isn’t good general purpose design… there’s an accessor for part. That returns a string instead of a number. JS doesn’t have an option type, so if it isn’t set then would be null, which would coerce to 0. We’d probably go with option, and then to implement in JS you’d need a tiny bit of glue code.
+
+**George Kulakowski:** Yes, I do think I mentioned this would err on the side of typed. Someone was asking about why not more closely hewed to browser. Using strings wouldn’t help because you still have to go to your native implementation.
+
+**Lin Clark:** Any more questions?
+
+Ok, consensus poll. Anyone object? [long awkward silence]
+
+Great, we have a new API!
```

Comment:
Why not align with the existing `URL` spec rather than make a new spec?

https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
The idea should be that binding for one language work the same regardless if using `URL` in a browser, or `URL` in WASI. It seems like a good idea for developer consistency.

Reply:
I see discussion regarding this above, but still. 

Or is the plan that WASI will automatically map to URL behind the scenes, and people will only write for WASI and it works everywhere?